File: BOJ/완전탐색/2309/2309.py
# 2309 일곱 난쟁이
# https://www.acmicpc.net/problem/2309
import sys
input = lambda : sys.stdin.readline().rstrip()

heights = [] # 키 리스트

for _ in range(9):
    heights.append(int(input()))

# 완전탐색
for i in range(9):
    for j in range(i+1,9):
        # 다행스럽게도 일곱 난쟁이의 키의 합이 100
        if sum(heights) - heights[i] - heights[j] == 100:
            heights[i] = 0
            heights[j] = 0
            break

heights.sort() # 정렬

# 출력
for h in heights:
    if h == 0:
        continue
    print(h)

File: BOJ/완전탐색/2309/2309.cpp
// 1654 일곱 난쟁이
// https://www.acmicpc.net/problem/2309
#include <iostream>
#include <algorithm>

using namespace std;

int main(void) {
    int heights[9];

    for (int i = 0; i <9; ++i) {
        scanf("%d",&heights[i]);
    }

    sort(heights, heights+9);

    int heights_Sum = 0, flag = 0;
    for (int &h: heights) heights_Sum += h;

    for (int i = 0; i < 8; ++i) {
        for (int j = i+1; j <9; ++j) {
            if (heights_Sum - heights[i] - heights[j] == 100) {
                flag = 1;
                heights[i] = -1;
                heights[j] = -1;
            }
        }

        if (flag) break;
    }

    for (int &h: heights) {
        if (h == -1) continue;
        printf("%d\n", h);
    }

    return 0;
}